extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/AwsHttpContextProducers.java

```diff
@@ -0,0 +1,65 @@
+package io.quarkus.amazon.lambda.http;
+
+import java.util.Collections;
+import java.util.Map;
+
+import javax.enterprise.context.RequestScoped;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.CDI;
+import javax.inject.Singleton;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayV2HTTPEvent;
+
+import io.quarkus.vertx.http.runtime.CurrentVertxRequest;
+import io.quarkus.vertx.http.runtime.QuarkusHttpHeaders;
+import io.vertx.core.MultiMap;
+import io.vertx.ext.web.RoutingContext;
+
+@Singleton
+public class AwsHttpContextProducers {
+
+    @RequestScoped
+    @Produces
+    public Context getAwsContext() {
+        return (Context) getContextObjects().get(Context.class);
+    }
+
+    @RequestScoped
+    @Produces
+    public APIGatewayV2HTTPEvent getHttpEvent() {
+        return (APIGatewayV2HTTPEvent) getContextObjects().get(APIGatewayV2HTTPEvent.class);
+    }
+
+    @RequestScoped
+    @Produces
+    public APIGatewayV2HTTPEvent.RequestContext getHttpRequestContext() {
+        return (APIGatewayV2HTTPEvent.RequestContext) getContextObjects().get(APIGatewayV2HTTPEvent.RequestContext.class);
+    }
+
+    private Map<Class<?>, Object> getContextObjects() {
+        Instance<CurrentVertxRequest> select = CDI.current().select(CurrentVertxRequest.class);
```

Comment:
Why not just inject this instead of looking it up each time? There is a performance cost associated with the lookup.

Reply:
You mean cache Instance<CurrentVertxRequest> right?  Ok, I added this.